### DISCUSSION TITLE
Compilation issue in Visual Studio

### DISCUSSION THREAD

**Discussion by guidofranceschini** [curl/curl#11367] [Q&A] [answered]
Hi,

I have downloaded curl 8.1.2 (and also cloned the GIT repository).
BTW: note that the zip file for 8.1.2 does not include the *.vcxproj files (but includes the *.vcxproj.filter files).

Unfortunately, compilation of libcurl 8.1.2 fails because the files lib/h2h3.c|h are missing. Those files are missing in GIT too.

If I remove the lib/h2h3.c|h files from the project, I get a few unresolved symbols (I use the Debug DLL configuration).
Here are the relevant output errors from Visual Studio 2022. It’s Italian, but should be easy to understand.

I hope you can help
Guido

1>Generazione del codice in corso...
1>   Creazione della libreria C:\TUPSNAP\GIT_CURL\build\Win32\VC14.30\DLL Debug\libcurld.lib e dell'oggetto C:\TUPSNAP\GIT_CURL\build\Win32\VC14.30\DLL Debug\libcurld.exp
1>cf-socket.obj : error LNK2019: riferimento al simbolo esterno _Curl_bufq_init non risolto nella funzione _cf_socket_ctx_init
1>cf-socket.obj : error LNK2019: riferimento al simbolo esterno _Curl_bufq_reset non risolto nella funzione _nw_in_read
1>cf-socket.obj : error LNK2019: riferimento al simbolo esterno _Curl_bufq_free non risolto nella funzione _nw_in_read
1>cf-socket.obj : error LNK2019: riferimento al simbolo esterno _Curl_bufq_is_empty non risolto nella funzione _do_connect
1>cf-socket.obj : error LNK2019: riferimento al simbolo esterno _Curl_bufq_read non risolto nella funzione _do_connect
1>cf-socket.obj : error LNK2019: riferimento al simbolo esterno _Curl_bufq_slurp non risolto nella funzione _do_connect
1>connect.obj : error LNK2019: riferimento al simbolo esterno _Curl_cf_haproxy_insert_after non risolto nella funzione _cf_he_insert_after
1>curl_log.obj : error LNK2001: simbolo esterno _Curl_cft_h1_proxy non risolto
1>curl_log.obj : error LNK2001: simbolo esterno _Curl_cft_haproxy non risolto
1>http.obj : error LNK2019: riferimento al simbolo esterno _Curl_dynhds_init non risolto nella funzione _Curl_http_req_make
1>http.obj : error LNK2019: riferimento al simbolo esterno _Curl_dynhds_free non risolto nella funzione _Curl_http_req_free
1>http.obj : error LNK2019: riferimento al simbolo esterno _Curl_dynhds_reset non risolto nella funzione _Curl_http_req_to_h2
1>http.obj : error LNK2019: riferimento al simbolo esterno _Curl_dynhds_count non risolto nella funzione _Curl_http_req_to_h2
1>http.obj : error LNK2019: riferimento al simbolo esterno _Curl_dynhds_set_opts non risolto nella funzione _Curl_http_req_to_h2
1>http.obj : error LNK2019: riferimento al simbolo esterno _Curl_dynhds_getn non risolto nella funzione _Curl_http_req_to_h2
1>http.obj : error LNK2019: riferimento al simbolo esterno _Curl_dynhds_get non risolto nella funzione _Curl_http_req_to_h2
1>http.obj : error LNK2019: riferimento al simbolo esterno _Curl_dynhds_add non risolto nella funzione _Curl_dynhds_add_custom
1>http_proxy.obj : error LNK2019: riferimento al simbolo esterno _Curl_cf_h1_proxy_insert_after non risolto nella funzione _Curl_cf_http_proxy_insert_after

**Comment by bagder** [comment, marked as answer]
> BTW: note that the zip file for 8.1.2 does not include the *.vcxproj files

Then you downloaded something else than the curl release zip file: https://curl.se/download/curl-8.1.2.zip

This contains several vcxproj files.

> Unfortunately, compilation of libcurl 8.1.2 fails because the files lib/h2h3.c|h are missing

No build uses those files anymore since fc2f1e547 (since release 8.1.0).

> I get a few unresolved symbols

It is not clear to me what you tried to build nor how.

**Reply by guidofranceschini** [reply]
Thanks for the prompt reply. You solved my problem.
Indeed, I downloaded curl-curl-8_1_2.zip, which is labeled as source code (zip)

Guido

**Reply by bagder** [reply]
Unfortunately we cannot make GitHub *not* provide those misleading archives.
